Modules/System Tests/Barcode/src/Dynamics2D/Dynamics2DQrCodeTest.Codeunit.al

```diff
@@ -0,0 +1,77 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 135060 "Dynamics QR Code Test"
+{
+    Subtype = Test;
+    TestPermissions = Disabled;
+
+    var
+        Assert: Codeunit "Library Assert";
+        SampleTxt: Label 'Hello, World!', Locked = true;
+        MaxCapacityLengthTxt: Label 'AaBbCcDdEe', Locked = true;
+
+    [Test]
+    procedure StringToQRCodeTest()
+    var
+        BarcodeImageProvider2D: Enum "Barcode Image Provider 2D";
+        IBarcodeImageProvider2D: Interface "Barcode Image Provider 2D";
+    begin
+        // [SCENARIO] A string variable is converted to qr code
+        IBarcodeImageProvider2D := BarcodeImageProvider2D::Dynamics2D;
+
+        // [WHEN] The string is converted
+        // [THEN] no error occured
+        IBarcodeImageProvider2D.EncodeImage(SampleTxt, Enum::"Barcode Symbology 2D"::"QR-Code");
```

Comment:
Another way to test that no error occurred is to clear the last error before the invocation of `EncodeImage` and then assert it is empty after the invocation.

Reply:
Done